### PR TITLE
 More using of referencies instead of values for Frect struct 

### PR DESCRIPTION
**Pull request by xrombik**

### CODE REVIEW

File: src/xrUICore/Windows/UIFrameWindow.cpp [hunk 1]
```diff
@@ -303,28 +303,28 @@ void CUIFrameWindow::draw_tile_line(Frect rect, int i, bool b_horz, Fvector2 con
         while (rect.lt.x + EPS_L < rect.rb.x)
         {
             get_points(rect, i, LTp, RBp, LTt, RBt);
-            rect.lt.x = RBp.x;
```
Comment (Kaffeine) [inline src/xrUICore/Windows/UIFrameWindow.cpp:306]: This change is not described in the PR. Is this (and similar below) move intentional?

File: src/xrGame/ui/UIMap.h [hunk 2]
```diff
@@ -54,8 +54,8 @@ class CUICustomMap : public CUIStatic, public CUIWndCallback
     virtual void Draw();
     virtual void Update();
     virtual void SendMessage(CUIWindow* pWnd, s16 msg, void* pData);
-    virtual bool IsRectVisible(Frect r);
-    virtual bool NeedShowPointer(Frect r);
+    virtual bool IsRectVisible(const Frect& r);
+    virtual bool NeedShowPointer(const Frect& r);
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIMap.h:58]: `r` is being changed inside the function and it is wrong to use references here.

File: src/xrGame/ui/UIMap.h [hunk 3]
```diff
@@ -54,8 +54,8 @@ class CUICustomMap : public CUIStatic, public CUIWndCallback
     virtual void Draw();
     virtual void Update();
     virtual void SendMessage(CUIWindow* pWnd, s16 msg, void* pData);
-    virtual bool IsRectVisible(Frect r);
-    virtual bool NeedShowPointer(Frect r);
+    virtual bool IsRectVisible(const Frect& r);
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIMap.h:57]: `r` is being changed inside the function and it is wrong to use references here.

**Comment by Xottab-DUTY** [comment]
Closed due to changes resulting in altered behavior.

**Pull request closed without merging**